### PR TITLE
[fix] Allow to extract `link` files, fixes #37

### LINKED ISSUE #37: Support for hard links
As indicated at https://github.com/mafintosh/prebuild/issues/110 `tar-fs` needs to have support for hard links, and ideally it would need to add support for the remaining file types supported by `tar-stream`.

### PR DESCRIPTION
**Pull request by piranna**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -265,15 +272,18 @@ exports.extract = function (cwd, opts) {
...
-      if (header.type === 'symlink') return onlink()
 
-      if (header.type !== 'file') {
-        if (strict) return next(new Error('unsupported type for ' + name + ' (' + header.type + ')'))
-        stream.resume()
-        return next()
+      switch (header.type) {
+        case 'file'   : return onfile()
+        case 'link'   : return onlink()
+        case 'symlink': return onsymlink()
```
Comment (mafintosh) [inline index.js]: for the link and symlink cases you need to resume to stream since the methods don't do that

Comment (piranna) [inline index.js]: Previously for the `symlink` it was not being resumed, nor here or at the `onlink` function. Was it a bug? Should I add a resume on `onlink` and `onsymlink` functions?

Comment (mafintosh) [inline index.js]: Oh you're right! we don't need to resume non-file streams since they are always empty. My bad.

**Comment by mafintosh** [comment]
There is a couple of `standard` style errors. Could you fix those (the if should be one-line and no white-space padding in the switch)? Other than that LGTM

**Comment by mafintosh** [comment]
Btw, do you think `xfs.unlink(name, ...` should be `xfs.unlink(path.resolve(cwd, name)` instead? It looks like a bug to me, although it was already in there before your PR. One of the arguments to xfs.symlink should probably be resolved as well (i can never remember which one)

**Comment by mafintosh** [comment]
Oh nevermind my previous comment. I see we already resolve name.

**Comment by piranna** [comment]
There you go :-) Please tell me if there's any other problem with this :-)

**Comment by mafintosh** [comment]
Thanks! Released in 1.12.0

**Comment by piranna** [comment]
Thanks to you! :-D By removing the archive check on prebuild and some
tune-ups now it works correctly in my case, I'm going to add the checks for
upload and later do a pull-request with the changes :-)

**Pull request merged**